File: src/main/java/org/example/stack/DailyTemps.java
package org.example.stack;

import java.util.Stack;

public class DailyTemps {

    public int[] dailyTemperatures(int[] temperatures) {
        int[] res = new int[temperatures.length];
        /*Stack<<Integer, Integer>> stack = new Stack<>();

        for (int i=0; i < temperatures.length; i++) {
            if (!stack.isEmpty() && stack.peek().fst < temperatures[i]) {
                int index = stack.pop().snd;
                res[index] = i - index;
            } else {
                stack.push(new Pair<>(temperatures[i], i));
            }
        }*/

        return res;
    }
}

File: src/main/java/org/example/stack/DailyTemps.py
from typing import List

# Monotonically decreasing stack
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        res = [0] * len(temperatures)
        stack = []  # pair: [temp, index]

        for i, t in enumerate(temperatures):
            while stack and t > stack[-1][0]:
                temp, index = stack.pop()
                res[index] = i - index
            stack.append([t, i])
        return res
